domain/registry/verifiable_credential.go

```diff
@@ -0,0 +1,46 @@
+package registry
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/nuts-foundation/nuts-demo-ehr/client"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+)
+
+type VerifiableCredentialRegistry interface {
+	CreateAuthorizationCredential(ctx context.Context, purposeOfUse, issuer, subjectID string, resources []credential.Resource) error
+}
+
+type verifiableCredentialRegistry struct {
```

Comment:
you could name this `httpVerifiableCredentialRegistry` since it's an HTTP implementation